providers/implementations/signature/rsa_sig.c

```diff
@@ -79,14 +87,41 @@ typedef struct {
     RSA *rsa;
     int operation;
 
+    /*
+     * Flag to determine if a full sigalg is run (1) or if a composable
+     * signature algorithm is run (0).
+     *
+     * When a full sigalg is run (1), this currently affects the following
+     * other flags, which are to remain untouched after their initialization:
+     *
+     * - flag_allow_md (initialized to 0)
+     */
+    unsigned int flag_sigalg : 1;
     /*
      * Flag to determine if the hash function can be changed (1) or not (0)
      * Because it's dangerous to change during a DigestSign or DigestVerify
      * operation, this flag is cleared by their Init function, and set again
      * by their Final function.
+     * Implementations of full sigalgs (such as RSA-SHA256) hard-code this
+     * flag to not allow changes (0).
      */
     unsigned int flag_allow_md : 1;
     unsigned int mgf1_md_set : 1;
+    /*
+     * Flags to say what are the possible next external calls in what
+     * consitutes the life cycle of an algorithm.  The relevant calls are:
+     * - init
+     * - update
+     * - final
+     * - oneshot
+     * All other external calls are regarded as utilitarian and are allowed
+     * at any time (they may be affected by other flags, like flag_allow_md,
+     * though).
+     */
+    unsigned int flag_allow_init : 1;
```

Comment:
This flag seems superfluous to me as you can always reinitialize.

Reply:
Dropped